### PR TITLE
core(time-to-first-byte): use receiveHeadersStart

### PR DESCRIPTION
**Pull request by connorjclark**
I added `receiveHeadersStart` to network timings in https://chromium-review.googlesource.com/c/chromium/src/+/4556570 . It will be available in M116.

This PR changes ttfb calculation to use `timing.receiveHeadersStart` instead of `timing.receiveHeadersEnd`. Using the end time potentially misreports ttfb in uncommon cases where the server either intentionally delays some headers (say, it push out some link headers early then does the rest after some thinking); or sends so many headers that they arrive over multiple packets. In either case, we care about the network connection latency, so we want to exclude such delays.

### CODE REVIEW

**Review by adamraine — COMMENTED** [review]
This is good. Should we update the usages of `receiveHeadersEnd` in `server-reponse-time.js` and `network-analyzer.js` as well?

**Comment by connorjclark** [comment]
> Should we update the usages of receiveHeadersEnd in server-reponse-time.js

As the `server-response` audit is documented, yeah we should change it there too. It seems the intention was to be a synonym for ttfb.

Without considering the current state of the audit, I'd consider "server response" to be pretty ambiguous, you could argue it can be the first byte of the headers or the first byte of the response body. I don't have a good argument for one way or the other, except that: if it was meant to be the first byte of the headers, why wasn't the audit called TTFB?

> `network-analyzer.js`?

`transferSize` includes bytes from the header, so I think changing that too is right. good catch.

**Comment by connorjclark** [comment]
Let's defer the network analyzer changes until after we update the Lantern database. #15150

**Comment by brendankenny** [comment]
What other uses are there? Would it be helpful to make a tracking bug to make sure they all get moved over?

**Review by adamraine — APPROVED** [review]

File: core/lib/network-request.js [hunk 1]
```diff
@@ -449,6 +452,19 @@ class NetworkRequest {
...
+   * `timing.receiveHeadersStart` was added recently, and will be in M116. Until then,
+   * set it to receiveHeadersEnd, which is close enough, to allow consumers of NetworkRequest
+   * to use the new field without accounting for this backcompat.
+   */
+  _backfillReceiveHeaderStartTiming() {
+    // Do nothing if a value is already present!
+    if (!this.timing || this.timing.receiveHeadersStart !== undefined) return;
+
+    this.timing.receiveHeadersStart = this.timing.receiveHeadersEnd;
+  }
```
Comment (adamraine) [inline core/lib/network-request.js:466]: Add some test cases for this?

**Pull request merged**